test/framework/src/main/java/org/elasticsearch/test/AbstractQueryTestCase.java

```diff
@@ -413,7 +412,9 @@ public final void testUnknownObjectException() throws IOException {
                     // Parse the valid query and inserts a new object level called "newField"
                     XContentParser.Token token;
                     while ((token = parser.nextToken()) != null) {
-                        if (token == XContentParser.Token.START_OBJECT) {
+                        if (token == XContentParser.Token.START_ARRAY) {
```

Comment:
@cbuescher Would you mind checking the changes in the file out? I had to change this to make the `testUnknownObjectException()` for the percolator query builder to work.

The percolate query builder now allows multiple documents to percolate via an object array.
Since any content is allowed under this array, this was a suffle protected field. However
these fields were not picked up once this test logic got inside an array. This change should fix that.

Reply:
@martijnvg Took me a bit to get back into understanding what this weird test does, but now it looks good (at least the changes in this file and the adding of "documents" to the exception list in PercolateQueryBuilderTests). I assume this works well with the other tests AbstractQueryTestCase.

Reply:
@cbuescher Thanks for checking. Yes, this change does also work with other tests (`gradle core::test` passed).